server/src-lib/Hasura/Server/Init.hs

```diff
@@ -1084,23 +1085,13 @@ serveOptionsParser =
 -- and catalog schema versions.
 downgradeShortcuts :: [(String, String)]
 downgradeShortcuts = 
-  [ ("v1.0.0-beta.1", "16")
-  , ("v1.0.0-beta.2", "17")
-  , ("v1.0.0-beta.3", "17")
-  , ("v1.0.0-beta.4", "19")
-  , ("v1.0.0-beta.5", "19")
-  , ("v1.0.0-beta.6", "22")
-  , ("v1.0.0-beta.7", "24")
-  , ("v1.0.0-beta.8", "26")
-  , ("v1.0.0-beta.9", "26")
-  , ("v1.0.0-beta.10", "27")
-  , ("v1.0.0-rc.1", "28")
-  , ("v1.0.0", "28")
-  , ("v1.1.0-beta.1", "29")
-  , ("v1.1.0-beta.2", "30")
-  , ("v1.1.0-beta.3", "31")
-  , ("v1.1.0", "31")
-  ]
+  $(do addDependentFile "catalog_versions"
+       s <- runIO (readFile "catalog_versions")
```

Comment:
This is a minor point, but I think you can nest splices, so you could use `Data.FileEmbed.embedStringFile` here to avoid having to call `addDependentFile` manually (since we already depend on `file-embed` anyway).

Reply:
`embedStringFile` produces an `Exp`, but I need the string so that I can parse it.

Reply:
Right, but I think you can nest splices, so you could do:

```haskell
$(let s = $(embedStringFile "catalog_versions")
      parseVersions = ...
  in ...)
```

Reply:
I see, thanks.